src/error.rs

```diff
@@ -213,26 +267,59 @@ pub(crate) fn map_error(input: rustls::Error) -> rustls_result {
     match input {
         Error::InappropriateMessage { .. } => InappropriateMessage,
         Error::InappropriateHandshakeMessage { .. } => InappropriateHandshakeMessage,
-        Error::CorruptMessage => CorruptMessage,
-        Error::CorruptMessagePayload(_) => CorruptMessagePayload,
+
         Error::NoCertificatesPresented => NoCertificatesPresented,
         Error::DecryptError => DecryptError,
-        Error::PeerIncompatibleError(_) => PeerIncompatibleError,
-        Error::PeerMisbehavedError(_) => PeerMisbehavedError,
+        Error::PeerIncompatible(_) => PeerIncompatibleError,
+        Error::PeerMisbehaved(_) => PeerMisbehavedError,
         Error::UnsupportedNameType => UnsupportedNameType,
         Error::EncryptError => EncryptError,
 
+        Error::InvalidMessage(e) => match e {
+            InvalidMessage::HandshakePayloadTooLarge =>  MessageHandshakePayloadTooLarge,
+            InvalidMessage::InvalidCcs => MessageInvalidCcs,
+            InvalidMessage::InvalidContentType => MessageInvalidContentType,
+            InvalidMessage::InvalidCertificateStatusType(_) => MessageInvalidCertStatusType,
+            InvalidMessage::InvalidCertRequest => MessageInvalidCertRequest,
+            InvalidMessage::InvalidDhParams => MessageInvalidDhParams,
+            InvalidMessage::InvalidEmptyPayload => MessageInvalidEmptyPayload,
+            InvalidMessage::InvalidKeyUpdate(_) => MessageInvalidKeyUpdate,
+            InvalidMessage::InvalidServerName => MessageInvalidServerName,
+            InvalidMessage::MessageTooLarge => MessageTooLarge,
+            InvalidMessage::MessageTooShort => MessageTooShort,
+            InvalidMessage::MissingData(_) => MessageMissingData,
+            InvalidMessage::MissingKeyExchange => MessageMissingKeyExchange,
+            InvalidMessage::NoSignatureSchemes => MessageNoSignatureSchemes,
+            InvalidMessage::TrailingData(_) => MessageTrailingData,
+            InvalidMessage::UnexpectedMessage(_) => MessageUnexpectedMessage,
+            InvalidMessage::UnknownProtocolVersion => MessageUnknownProtocolVersion,
+            InvalidMessage::UnsupportedCompression => MessageUnsupportedCompression,
+            InvalidMessage::UnsupportedCurve(_) => MessageUnsupportedCurve,
+            InvalidMessage::UnsupportedKeyExchangeAlgorithm(_) => MessageUnsupportedCompression,
+            _ => General,
```

Comment:
I was surprised I needed this `_` arm since I thought the matches above were exhaustive, but the compiler complains otherwise. Maybe more investigation needed.

(Same for the `Error::InvalidCertificate(e) => match e` lower)

Reply:
The upstream `InvalidMessage` enum [has the #[non_exhaustive] attribute](https://github.com/rustls/rustls/blob/6831835c56696d61024f2ae1261b41edad54e685/rustls/src/error.rs#L97), so it's an [error to match against it in any match statement that lacks an `_`.](https://doc.rust-lang.org/reference/attributes/type_system.html).

Rather than bailing out to General, we should probably define a `MessageInvalidOther` for this case.